pptx_blueprint/__init__.py

```diff
@@ -87,6 +89,21 @@ def _find_shapes_in_slide(slide):
 
         return matched_shapes
 
+    def _get_all_shapes(self) -> Iterable[BaseShape]:
+        # Do we need all the shapes? Perhaps we should filter on tags here.
```

Comment:
We need to iterate over each shape at least once and apply the re.match to identify whether there's some work for us to do. It's a decision of style if you want to first get all shapes and then match and work, or if you get all matching slides and then do some work. For a (extremely) big presentation, however, generating a list of all slides may be quite memory expensive and possibly slow. Both issues could be eliminated if you'd just replace the square brackets with regular brackets, thereby changing the list comprehension into a generator expression.

Reply:
Also, please inline...

Reply:
> Also, please inline...
Do you mean the comment? Isn't it a bit long to be inlined? (PEP8 says "use inline comments sparingly ;) )

Reply:
Inlining means to eliminate the function and use the code of the function body instead of the function call. This is reasonable in this particular case because

- the function is only used once (and unlikely to be needed multiple times)
- the inline version is almost equally short and equally descriptive.
  instead of
  ~~~
  all_shapes = self._get_all_shapes()
  ~~~
  just use
  ~~~
  all_shapes = (shape
                for slide in self._presentation.slides
                for shape in slide.shapes)
  ~~~

Reply:
Thanks for clarifying my comment, Tim. I now see that I could have been a little more verbose :-)